Top100/top100songs.py

```diff
@@ -0,0 +1,257 @@
+    """
+    # Check input validation
+    if text is '':
+        return dict()
+    if type(text) is not str:
+        raise ValueError("Input must be string")
+
+    # Clean rows from Text
+    text.replace("\n", ' ')
+
+    # Insert word without punctuations
+    word_count = {}
+    for word in text.split():
+        word = word.strip(string.punctuation).lower()
+        try:
+            word_count[word] = word_count[word] + 1
+        except KeyError:
+            word_count[word] = 1
+    return word_count
+
+
+def get_top_100_songs() -> dict:
+    """This function gets the top 100 songs as published on the
+    Billboard hot-100 page (https://www.billboard.com/charts/hot-100)
+    :return: songs_dict: dict
+    """
+    url = 'https://www.billboard.com/charts/hot-100'
+
+    content = get_content_from_url(url)
+
+    chart_list = content.find_all('ul', class_="o-chart-results-list-row")
+
+    # Find all ranks, and correlating titles and artists
+    songs_dict = {}
+    for element in chart_list:
+        rank = element.find('span', class_='a-font-primary-bold-l').string.strip('\n\t')
+        song_title = element.find('h3', class_='a-no-trucate').string.strip('\n\t')
+        artist = element.find('span', class_='a-no-trucate').string.strip('\n\t')
+
+        lyrics = get_lyrics(song_title, artist)
+        if rank not in songs_dict:
+            songs_dict[rank] = {'Title': song_title,
+                                'Artist': artist.replace(' &', ','),
+                                'Word count': count_words_in_text(lyrics)}
+        else:
+            raise LookupError(f"parsing error. rank {rank} was already accounted for")
+
+    return songs_dict
+
+
+def merge_all_word_counts(song_word_count: dict, tot_word_count: dict) -> dict:
+    """Sums the word counts of all the various song_count dictionaries
+    :param song_word_count: dict
+    :param tot_word_count: dict
+    :return: tot_word_count: dict
+    """
+    for word in song_word_count.keys():
+        try:
+            tot_word_count[word] = tot_word_count[word] + song_word_count[word]
+        except KeyError:
+            tot_word_count[word] = song_word_count[word]
+    return tot_word_count
+
+
+def merge_dictionaries_by_key(songs_dict: dict, key: any) -> dict:
+    """This function runs through a dictionary with values of type dictionary,
+    and merges the values's values by the key given (value[key]).
+    :param songs_dict: dict, to each key, the value is of type(dict)
+    :param key: str, the key within the songs_dict value dictionary to be used (value[key])
+    :return: tot_word_count: dict
+    """
+    tot_word_count = dict()
+    for ranked_song_info in songs_dict.values():
+        tot_word_count = merge_all_word_counts(ranked_song_info[key], tot_word_count)
+    return tot_word_count
+
+
+def plot_most_used_words(tot_word_count: dict) -> plt:
+    """
+    This function gets a dictionary of word: counts of the top 100 ranked songs lyrics.
```

Comment:
dictionary of word:?

Reply:
changed documentation to both plot function to describe more accurately it's content.